sycl/plugins/esimd_emulator/pi_esimd_emulator.hpp

```diff
@@ -105,6 +105,27 @@ struct _pi_queue : _pi_object {
   cm_support::CmQueue *CmQueuePtr = nullptr;
 };
 
+struct cm_surface_ptr_t {
+  // 'UP' means 'User-Provided' in CM Lib - corresponding to
+  // Buffer/Image created with PI_MEM_FLAGS_HOST_PTR_USE option in
+  // SYCL
+  enum SurfaceType {
+    type_none,
+    type_regular_buffer,
+    type_user_provided_buffer,
+    type_regular_image,
+    type_user_provided_image
+  };
+  SurfaceType tag = type_none;
+
+  union {
+    cm_support::CmBuffer *RegularBufPtr = nullptr;
+    cm_support::CmBufferUP *UPBufPtr;
+    cm_support::CmSurface2D *RegularImgPtr;
+    cm_support::CmSurface2DUP *UPImgPtr;
+  };
+};
+
 struct _pi_mem : _pi_object {
   _pi_mem() = default;
```

Comment:
Question: is default construction intended for this class (I see it's only used as a base class and in all use cases non-default constructor is used)?